pod.go

```diff
@@ -389,6 +389,11 @@ func (p *Pod) Annotations(key string) (string, error) {
 	return value, nil
 }
 
+// SetOrAddAnnotation sets or adds an annotation
+func (p *Pod) SetOrAddAnnotation(key, value string) {
+	p.config.Annotations[key] = value
```

Comment:
I think you should save the PodConfig locally after you added a new Annotation. You can refer to this: https://github.com/containers/virtcontainers/blob/master/api.go#L424-L429
Indeed, depending on where you call this function, you may lose this Annotation if the config is not stored afterwards.
But this will cost a bit of performances because we will store every time we add a new annotation. I would suggest a second function in case you want to add/set several Annotations:
```go
func (p *Pod) SetAnnotations(annotationList map[string]string)
``` 
@sameo @jodh-intel what do you think ?